frontend/packages/metal3-plugin/src/models.ts

```diff
@@ -0,0 +1,14 @@
+import { K8sKind } from '@console/internal/module/k8s';
+
+export const BaremetalHostModel: K8sKind = {
```

Comment:
See my [comment on KubeVirt models](https://github.com/openshift/console/pull/1592#discussion_r290353110) - shouldn't we use `crd: true` here?

@spadgett In `kubevirt/web-ui` fork, any additional models are declared _without_ the `crd` flag, which is a mistake IMO.

Reply:
I agree, although when I add the crd flag, I am getting `kindObj: no model for kind BareMetalHost` in the console and the ListPage won't show up. It seems like there is a mismatch in keys of the resources in redux store and how crd resources are retrieved.

Reply:
Fixed the problem and added `crd: true` flag to the model.

Reply:
> It seems like there is a mismatch in keys of the resources in redux store and how crd resources are retrieved.

There's currently a duality in how Console references models. Non-CRD models are typically referenced only by their `kind`, CRD models are referenced via `referenceForModel` function that takes their `apiGroup`, `apiVersion` and `kind` into account.

Getting `kindObj: no model for kind BareMetalHost` could mean you aren't referencing your CRD model via `referenceForModel` function.